### PR TITLE
detach keys from lease

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: lease/lessor.go [hunk 1]
```diff
@@ -194,19 +198,22 @@ func (le *lessor) Grant(id LeaseID, ttl int64) (*Lease, error) {
 
 func (le *lessor) Revoke(id LeaseID) error {
 	le.mu.Lock()
-	defer le.mu.Unlock()
 
 	l := le.leaseMap[id]
 	if l == nil {
 		return ErrLeaseNotFound
```
Comment (heyitsanthony) [inline lease/lessor.go]: `Unlock`

Comment (xiang90) [inline lease/lessor.go]: will fix.

**Comment by heyitsanthony** [comment]
lgtm aside from missing unlock

**Comment by xiang90** [comment]
@heyitsanthony Fix the unlock issue. Will merge on green. Will write integration test to ensure this works correctly.

**Pull request merged**